compiler/src/dotty/tools/dotc/repl/ammonite/Protocol.scala

```diff
@@ -15,12 +15,20 @@ case class TermState(
 ) extends TermAction
 
 object TermState {
+  // Using unapply instead exposes a dotty/scalac variation. Because TermState
+  // is case class, scalac generate an unapply with this exact signature, that
+  // is used by the TermInfo | TermAction unapplies. With dotty, the generated
```

Comment:
I would say `TermInfo and TermAction` instead of `TermInfo | TermAction` to avoid confusion with union types here.